addons/pos_loyalty/models/pos_session.py

```diff
@@ -40,28 +40,51 @@ def _load_data_params(self, config_id):
             },
         })
 
+        # add missing product fields used in the reward_product_domain
+        missing_fields = self._get_reward_product_domain_fields(params) - set(params['product.product']['fields'])
+
+        if missing_fields:
```

Comment:
once there is an invalid field name here, don't proceed on extending the product.product fields to load. It'll be useless anyway.

Reply:
@caburj I modified the code.